packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -215,9 +205,18 @@ class DropDownButton<T> extends StatelessComponent {
   void _showDropDown(BuildContext context, int selectedIndex, GlobalKey indexedStackKey) {
     final RenderBox renderBox = indexedStackKey.currentContext.findRenderObject();
     final Rect rect = renderBox.localToGlobal(Point.origin) & renderBox.size;
+    final NavigatorState navigator = Navigator.of(context);
```

Comment:
That's what I'd tried at first. The problem is that the navigator is needed to size the menu as it's coming down but the route machinery clears the navigator before the dismiss animation is finished. Which results in this:
```
android: I/flutter : Exception caught while building _ModalScope-[GlobalKey 745801642]
android: I/flutter : The null object does not have a getter 'context'.
android: I/flutter : NoSuchMethodError: method not found: 'context'
android: I/flutter : Receiver: null
android: I/flutter : Arguments: []
android: I/flutter : Stack trace:
android: I/flutter : #0      Object._noSuchMethod (dart:core-patch/object_patch.dart:42)
android: I/flutter : #1      Object.noSuchMethod (dart:core-patch/object_patch.dart:45)
android: I/flutter : #2      _DropDownRoute.position (packages/flutter/src/material/dropdown.dart:164)
android: I/flutter : #3      _ModalScope.build (packages/flutter/src/widgets/routes.dart:254)
android: I/flutter : #4      StatusTransitionState.build (packages/flutter/src/widgets/status_transitions.dart:49)
android: I/flutter : #5      ComponentElement.performRebuild (packages/flutter/src/widgets/framework.dart:1070)
...
```